fix: make urls argument required and clean up clap configuration

uncenter opened the pull request:
Switches to using doc comments for the help strings of flags, makes the URLs argument required, and removes extra configuration for clap that is the default.

uncenter commented:
I've also now corrected the behavior of `urls` and `path` if I understand them correctly. You can pass as many URL arguments to the command to download, and if you want all the files to instead go in a specific directory you can add `-p dist/` or whatever. Previously:
- If I pass 1 URL followed by a directory path, the files are downloaded into the path of the directory specified. ✅
- If I pass 2 URLs, I would get a `No such file or directory (os error 2)` error (I believe it is trying to write the first URL to the path of the second). ❌ 
- If I passed more than 2 URL arguments, I would get `error: Found argument '<some url>' which wasn't expected, or isn't valid in this context`. ❌
- Due to the above two items, passing two or more URL arguments followed by an output path argument doesn't work (only one URL followed by an output path argument works). ❌
Now:
- If I pass 1 URL followed by a directory path, the files are downloaded into the path of the directory specified. ✅
- If I pass 2 URLs, both files/directories are downloaded correctly. ✅
- If I passed more than 2 URL arguments, all files/directories are downloaded correctly. ✅
- If I pass any number of URLs followed by `-p <path>`, all files from the URLs before are downloaded into the specified path. ✅

alok8bb commented:
Hi @uncenter, 

Thanks for your effort in creating this PR!

> If I pass 2 URLs, I would get a No such file or directory (os error 2) error (I believe it is trying to write the first URL to the path of the second, like the above list item). ❌
If I passed more than 2 URL arguments, I would get error: Found argument '<some url>' which wasn't expected, or isn't valid in this context. ❌

I believe there may be a misunderstanding about how multiple URLs are downloaded to a user-specified path. When there are two or more URLs, they're separated by commas. Downloading files/directories to custom paths was actually implemented in response to a [feature request](https://github.com/alok8bb/cloneit/issues/6).

Here are some examples:
1. To clone into the current directory without creating a new directory of the same name (which is the default behavior):
`cloneit <url> .`

2. To clone multiple directories/files into the same directory or a custom directory:
`cloneit <url1>,<url2> .` or `cloneit <url1>,<url2> custom_directory`

In the case where `<url1>` is a file and `<url2>` is a directory, this command would download the contents of the directory and the file from `<url1>`, placing them all inside either the current directory (`.`) or the specified `custom_directory`.

This is the same way how `git` handles it, so using a path flag might not be a good idea. The only current issue is that when the target directory or file already exists, cloneit returns an error stating that the file/directory already exists but I think it is okay that way.

alok8bb commented:
> The only current issue is that when the target directory or file already exists, cloneit returns an error stating that the file/directory already exists but I think it is okay that way.

I think that is fixed by https://github.com/alok8bb/cloneit/pull/11

uncenter commented:
> This is the same way how git handles it, so using a path flag might not be a good idea. 

Are you referring to `git clone`? This comparison isn't right because Git only accepts a single repository URL in that command followed by a path.

> I believe there may be a misunderstanding about how multiple URLs are downloaded to a user-specified path. When there are two or more URLs, they're separated by commas.

Hmmm that seems really counterintuitive.

alok8bb commented:
> Hmmm that seems really counterintuitive.

I forgot to put the examples in the README, but I think its fine or even better this way? 🤔

uncenter commented:
I'll leave that behavior as is for now, and maybe I'll open an issue later to try and convince you ahah. Reverted this PR to its prior state, I think making `urls` required though is still a good change but I can also split it into a separate PR if desired.

alok8bb commented:
Thanks for your contribution!
